lib/PluginBase.js

```diff
@@ -272,51 +203,37 @@ class PluginBase {
 
             let hostState;
             try {
-                hostState = await this.getStateAsync(hostNamespace + '.enabled');
+                hostState = await this.getState(`${hostNamespace}.enabled`);
             } catch {
                 // ignore
             }
             if (hostState && typeof hostState.val !== 'object' && hostState.val !== undefined) {
                 // We simply use the host enabled flag state
-                return this._initialize(pluginConfig, !!hostState.val, callback);
+                return this._initialize(pluginConfig, !!hostState.val);
             }
         }
 
-        this._initialize(pluginConfig, pluginConfig.enabled === undefined ? true : !!pluginConfig.enabled, callback);
+        this._initialize(pluginConfig, pluginConfig.enabled === undefined ? true : !!pluginConfig.enabled);
     }
 
     /**
      * @internal
      * @param {Record<string, any>} pluginConfig
      * @param {string | boolean} activate
-     * @param {import("@iobroker/plugin-base/types").InitCallback} callback
      */
-    async _initialize(pluginConfig, activate, callback) {
+    async _initialize(pluginConfig, activate) {
         if (activate) {
-            this.log.debug('Initialize Plugin (enabled=' + activate + ')');
+            this.log.debug(`Initialize Plugin (enabled=${activate})`);
             pluginConfig.enabled = activate;
-            this.init(pluginConfig, async (err, success) => {
-                try {
-                    if (!err && success) {
-                        await this.setActive(true);
-                    } else {
-                        await this.setActive(false);
-                    }
-                } catch (ex) {
-                    if (!err) {
-                        err = ex
-                    }
-                }
-                callback(err, success);
-            });
-        } else {
-            this.log.debug('Do not initialize Plugin (enabled=' + activate + ')');
             try {
+                await this.init(pluginConfig);
+                await this.setActive(true);
```

Comment:
code is not the same ... old code returned a boolean and only if that boolean is true (setActive is called with true ... so here better const success = await .... setActive(success)

... ok OORRR methiod is now throwing also in non error but unsuccesseffull cases ... also good, just needs to make clear in docs

Reply:
Yes, that was the idea see https://github.com/ioBroker/plugin-base/pull/126/files#diff-6d076f09c97ce21f2d0659eeed7e34aa20e6fccbad5311f6932b71e1b6c1a97dR39